### DISCUSSION TITLE
關於中文與英文、中文與數字間添加間距的疑問

### DISCUSSION THREAD

**Discussion by sgalal** [sparanoid/chinese-copywriting-guidelines#166] [General]
**在使用中，發現一些邊緣情況：**

1\. 如 #48, #102 所言，「2019 年 12 月 25 日」這樣的寫法使得日期距離太大，影響美觀。

2\. 如 #59 所言，專有名詞應當不空格，例如應當寫為「U盘」而不是「U 盘」。但「專有名詞」的判斷十分模糊，如「維生素C」、「Unicode聯盟」是否應當添加空格？

3\. （接第 2 條）在同一上下文中，很可能同時出現「來 M」和「M巾」，此時若判斷前者非專有名詞，而後者為專有名詞，就會出現排版上不一致的情況，影響美觀。

4\. 「豆瓣FM」等按照官方所定義的格式不添加空格，但上下文的其他短語添加空格，在某些字型下差異明顯，影響美觀。

**進一步地，對於在 HTML 中的排版，有以下疑問：**

1\. 有中文譯名的英文專有名詞，在中文環境下，是否應當標注為英文環境？

如：

```html
<p lang="zh-HK">使用 Google 搜尋</p>
```

還是：

```html
<p lang="zh-HK">使用 <span lang="en-US">Google</span> 搜尋</p>
```

2\. 沒有中文譯名的英文專有名詞，在中文環境下，是否應當標注為英文環境？

如：

```html
<p lang="zh-HK">使用 JavaScript 編寫</p>
```

還是：

```html
<p lang="zh-HK">使用 <span lang="en-US">JavaScript</span> 編寫</p>
```

3\. 中英混合的詞，在中文環境下，英文部分是否應當標注為英文環境？

如：

```html
<p lang="zh-CN">豆瓣FM</p>
```

還是：

```html
<p lang="zh-CN">豆瓣<span lang="en-US">FM</span></p>
```

**Comment by liyishuai** [comment]
#100 有提及，再次引述《中文出版物夹用英文的编辑规范》（中华人民共和国新闻出版行业标准CY/T 154—2017号）第8.1节：
> 中文文本中夹用英文时，应根据所选用的中英文字体、字符间距以及排版的视觉效果决定英文词句与中文文字之间是否留有空格间距。如留空格，应保证体例的统一。

据此标准，如果`豆瓣FM`不加空格，则体例应统一无空格。
我在https://github.com/sparanoid/chinese-copywriting-guidelines/pull/105#issuecomment-577149056 提的的建议是将空格列为有争议规则，据应用场景决定加还是不加，而非强制。
从长远角度，夹用英文的排版问题应当由渲染软件解决，而非通过编码。

**Comment by sgalal** [comment]
@liyishuai 

> 据此标准，如果豆瓣FM不加空格，则体例应统一无空格。

该标准也可能被这样解读：「若通过某种原则确定某些场合应当添加空格，某些场合不应当添加空格，这也是一种『体例』，故不存在『不统一』的情况」。

因此，此处引述国家标准是无效的。

> 从长远角度，夹用英文的排版问题应当由渲染软件解决，而非通过编码。

这相当于将「在何处添加空格」的问题留给了「渲染软件」的编写者，仍然没有解决问题。

**Comment by liyishuai** [comment]
> > 从长远角度，夹用英文的排版问题应当由渲染软件解决，而非通过编码。
> 
> 这相当于将「在何处添加空格」的问题留给了「渲染软件」的编写者，仍然没有解决问题。

不是“添加空格”，而是安排文字间距。我的观点：空格不是汉语的一部分，中文编码不应包含` 英 文 空 格`，只有排版时需要合理安排**文字间距**，正面例子如微信。

**Comment by sgalal** [comment]
@liyishuai 

> 不是“添加空格”，而是安排文字间距。

我重新叙述：这相当于将「如何安排文字间距」的问题留给了「渲染软件」的编写者，仍然没有解决问题。

**Comment by liyishuai** [comment]
所以我加了状语
> 从长远角度，

我不反对在渲染软件很难更新的场景出于视觉目的添加空格，但反对强制推行此规则。

**Comment by sgalal** [comment]
我希望讨论者理解我讨论的出发点是「中文与西文、中文与数字间，在不同场合下，是否需要留出一定文字间距，以带来最好的阅读体验」，而不是「是否一定要添加 U+20 字符」或者「是否一定要借助渲染软件实现」。

目前，在绝大多数情况下「需要留出一定文字间距」已经是共识，但是在某些边缘情况下仍然存在问题，我希望讨论的是「如何处理这些边缘情况」的问题。

**Comment by liyishuai** [comment]
感谢指出，我已意识到自己偏题。
在“大多数情况下‘需要留出一定文字间距’”之基础上讨论如何处理“某些加空格的写法间隔过大”，似乎在假定“文字间距即空格”，或“无空格则无间距”，我不知如何在此假设下讨论，导致前文出现轻度抬杠。
解决方案：严格区分空格与间距，将空格视为制造间距的一种手段，并充分认识到该手段可能引发的后果（例如间距过大，不如无间距）。

**Comment by KagurazakaHanabi** [comment]
提供个 App Store 的例子作为参考：
![IMG_7833(20200205-172444)](https://user-images.githubusercontent.com/23377765/73828633-9992e500-483c-11ea-8259-4de4d1ea7be4.PNG)
